templates/company-board/test/company-board.js

```diff
@@ -111,170 +116,203 @@ contract('Company with board', ([_, owner, boardMember1, boardMember2, shareHold
-      assert.equal(installedApps.voting.length, 2, 'should have installed 2 voting apps')
-      assert.equal(installedApps['token-manager'].length, 2, 'should have installed 2 token manager apps')
-      acl = ACL.at(await dao.acl())
-      agent = Agent.at(installedApps.agent[0])
-      finance = Finance.at(installedApps.finance[0])
-      boardVoting = Voting.at(installedApps.voting[0])
-      shareVoting = Voting.at(installedApps.voting[1])
-      boardTokenManager = TokenManager.at(installedApps['token-manager'][0])
-      shareTokenManager = TokenManager.at(installedApps['token-manager'][1])
-    })
+    const itHandlesInstanceCreationsProperly = (useAgentAsVault) => {
+      // Test when the organization is created with an Agent app or a Vault app
+      
+      before('build dao ID', () => {
+        daoID = randomId()
+      })
 
-    it('costs ~10.4e6 gas', async () => {
-      const prepareGas = prepareReceipt.receipt.gasUsed;
-      const setupGas = setupReceipt.receipt.gasUsed;
-      const totalGas = prepareGas + setupGas;
-      assert.isAtMost(prepareGas, 5.0e6, 'prepare script should cost almost 5.0e6 gas')
-      assert.isAtMost(setupGas, 5.4e6, 'setup script should cost almost 5.4e6 gas')
-      assert.isAtMost(totalGas, 10.4e6, 'prepare + setup scripts should cost almost 10.4e6 gas');
-    })
+      before('create company entity', async () => {
+        prepareReceipt = await template.prepareInstance(SHARE_TOKEN_NAME, SHARE_TOKEN_SYMBOL, { from: owner })
+        setupReceipt = await template.setupInstance(
+          daoID, 
+          BOARD_MEMBERS, 
+          SHARE_HOLDERS, 
+          SHARE_STAKES, 
+          [BOARD_VOTE_DURATION, BOARD_SUPPORT_REQUIRED, BOARD_MIN_ACCEPTANCE_QUORUM],
+          [SHARE_VOTE_DURATION, SHARE_SUPPORT_REQUIRED, SHARE_MIN_ACCEPTANCE_QUORUM],
+          DEFAULT_FINANCE_PERIOD,
+          useAgentAsVault,
+          { from: owner }
+        )
+
+        dao = Kernel.at(getEventArgument(prepareReceipt, 'DeployDao', 'dao'))
+        boardToken = MiniMeToken.at(getEventArgument(prepareReceipt, 'DeployToken', 'token', 0))
+        shareToken = MiniMeToken.at(getEventArgument(prepareReceipt, 'DeployToken', 'token', 1))
+      })
 
-    it('registers a new DAO on ENS', async () => {
-      const ens = ENS.at((await deployedAddresses()).registry)
-      const aragonIdNameHash = namehash(`${daoID}.aragonid.eth`)
-      const resolvedAddress = await PublicResolver.at(await ens.resolver(aragonIdNameHash)).addr(aragonIdNameHash)
-      assert.equal(resolvedAddress, dao.address, 'aragonId ENS name does not match')
-    })
+      before('load apps', async () => {
+        const installedApps = getInstalledAppsById(setupReceipt)
+        if(useAgentAsVault) assert.equal(installedApps.agent.length, 1, 'should have installed 1 agent app')
+        else assert.equal(installedApps.vault.length, 1, 'should have installed 1 vault app')
```

Comment:
A quick note on style, we generally use the form of:

```
if (condition) {
  ...
} else {
  ...
}
```

Unless the clause is _really_ simple.

In any case, at some point we will have prettier handle most of this formatting :).